.mci.yml

```diff
@@ -744,6 +734,22 @@ buildvariants:
           - name: compile_and_test_with_shared_libs
           - name: compile_and_test_with_static_libs
 
+    # Test that we actually support building against the minium required C version claimed.
+    - name: ubuntu1804-min-libmongoc
+      display_name: "Ubuntu 18.04 with minimum libmongoc (MongoDB 4.4)"
+      batchtime: 1440 # 1 day
+      expansions:
+          build_type: "Debug"
+          tar_options: *linux_tar_options
+          cmake_flags: *linux_cmake_flags
+          mongodb_version: *version_latest
+          mongoc_version: "1.17.0-beta2"
```

Comment:
With this, the intent is if we introduce API into libmongoc, and start relying on that API in C++, we'll be alerted with a test failure if we do not bump the minimum C driver requirement.